### DISCUSSION TITLE
Homekit switch show active when Vacuum is on

### DISCUSSION THREAD

**Discussion by sushenmathur** [merdok/homebridge-miot#262] [Q&A] [answered]
Hi,
I have a Dreame L10 Pro (dreame.vacuum,p2029)
I have a action button in homekit for start clean which is displayed as a switch, but the switch state turn to false when the vaccum starts.
Can i create a action button/property control to display the swictch state as true when the vaccumm is running and turn to false when its stops and charging?

Thanks in advance for your help

**Comment by merdok** [comment, marked as answer]
The action switches are stateless buttons, meaning they do not have a state and turn themselves immediately off. This is due to the fact that actions in the miot protocol do not have a state.
But you already have the main vacuum switch which should turn on as soon as you start the vacuum  no matter from where the start command came.

**Comment by sushenmathur** [comment]
Thanks @merdok
I dont see a main swiitch in jomekit, just the action buttons that I've added in the config.
Do i need to do any thing else to get the main switch

**Reply by merdok** [reply]
Huh? I am pretty certain there should be one which is called the same name as you named your accessory in homebridge. There should also be a presence sensor. Did you maybe set your device to a customAccessory? That would indeed remove all default items.

**Reply by sushenmathur** [reply]
Oh yes looks like that was the issue the my custom accessory was set to true. Changed it now and I see a switch for the vacuum.
Thanks heaps
